### PR TITLE
Parallel Processing

### PR DESCRIPTION
**Pull request by howardbaek**
## What I did

Use `future::future()` to launch multiple R sessions (workers) and offload the slow processing to the `future` worker sessions.

## How I tested

I tested this by drawing inspiration from this Posit Community [post](https://community.rstudio.com/t/future-with-plumber/94477/2). 

Here is what I did:

```
library(plumber)
pr("plumber.R") %>% pr_run(port = 8080)
```

Put some dummy inputs (`link`, `service`, `model_name`, `vocoder_name`) into the Swagger UI which will output a Request URL. For ex, `http://127.0.0.1:9854/generate_from_gs?link=https%3A%2F%2Fdocs.google.com%2Fpresentation%2Fd%2F1Dw_rBb1hySN_76xh9-x5J2dWF_das9BAUjQigf2fN-E%2Fedit%3Fusp%3Dsharing&service=coqui&model_name=jenny&vocoder_name=jenny`. 

Create a txt file with urls:

```
# From within terminal
cat > urls.txt << EOF
http://127.0.0.1:9854/generate_from_gs?link=https%3A%2F%2Fdocs.google.com%2Fpresentation%2Fd%2F1Dw_rBb1hySN_76xh9-x5J2dWF_das9BAUjQigf2fN-E%2Fedit%3Fusp%3Dsharing&service=coqui&model_name=jenny&vocoder_name=jenny
http://127.0.0.1:9854/generate_from_gs?link=https%3A%2F%2Fdocs.google.com%2Fpresentation%2Fd%2F1Dw_rBb1hySN_76xh9-x5J2dWF_das9BAUjQigf2fN-E%2Fedit%3Fusp%3Dsharing&service=coqui&model_name=jenny&vocoder_name=jenny
EOF
```

Code to test plumber:

```
# From within terminal
# Take all urls and execute curl in parallel with 2 workers
cat urls.txt | xargs -P 2 -n 1 curl
```

References:
- https://posit.co/resources/videos/plumber-and-future-async-web-apis/
- https://community.rstudio.com/t/future-with-plumber/94477/2

### CODE REVIEW

**Comment by howardbaek** [comment]
@cansavvy This is ready now to be reviewed/merged.

**Comment by cansavvy** [comment]
Thanks for the really great PR description. This seems *promising* GET IT? 😆

File: inst/extdata/plumber.R [hunk 1]
```diff
@@ -1,5 +1,7 @@
 # Packages ----
 library(plumber)
+library(future)
+plan(multisession, workers = 25)
```
Comment (cansavvy) [inline inst/extdata/plumber.R:4]: What does the `workers` parameter do and why 25?

File: inst/extdata/plumber.R [hunk 2]
```diff
@@ -24,21 +26,23 @@ function(link, service, model_name, vocoder_name){
   # Temporary file
   tmp_video <- tempfile(fileext = ".mp4")
 
-  # Speaker Notes
-  pptx_path <- gsplyr::download(link, type = "pptx")
-  pptx_notes_vector <- ptplyr::extract_notes(pptx_path)
-  # Images
-  pdf_path <- gsplyr::download(link, type = "pdf")
-  image_path <- ptplyr::convert_pdf_png(pdf_path)
+  future::future({
```
Comment (cansavvy) [inline inst/extdata/plumber.R:29]: COOL!

**Review by cansavvy — COMMENTED** [review]
One question! But otherwise this seems worth a try. We should see if there's a way we can make a development branch or other type of more formal beta test before we deploy for real but that can be a *future* problem.

**Review by cansavvy — APPROVED** [review]

**Comment by howardbaek** [comment]
> One question! But otherwise this seems worth a try. We should see if there's a way we can make a development branch or other type of more formal beta test before we deploy for real but that can be a _future_ problem.

This is exactly what we need: https://edgararuiz.github.io/callthat/

File: inst/extdata/plumber.R (hunk 1 above)
Comment (howardbaek) [reply inst/extdata/plumber.R:4]: `workers` is the number of parallel processes to use. See https://future.futureverse.org/reference/multisession.html

**Pull request merged**